runtime/src/main/java/org/corfudb/runtime/view/replication/ChainReplicationProtocol.java

```diff
@@ -84,7 +86,7 @@ public ILogData peek(long globalAddress) {
      * @param data          The data to propagate, or NULL,
      *                      if it is to be a hole.
      */
-    protected void propagate(long globalAddress, @Nullable ByteBuf data) {
+    protected void propagate(long globalAddress, @Nullable ILogData data) {
         int numUnits = layout.getSegmentLength(globalAddress);
 
         for (int i = 1; i < numUnits; i++) {
```

Comment:
good point. it should be easy to fix -- let proparte take a start-index parameter.